faculty/clients/experiment.py

```diff
@@ -245,6 +334,98 @@ def make_restore_runs_response(self, data):
         return RestoreExperimentRunsResponse(**data)
 
 
+class SingleFilterValueField(fields.Field):
+    """
+    Field that serialises/deserialises a run filter.
+    """
+
+    def _is_valid_uuid(self, value, obj):
+        return isinstance(value, uuid.UUID) and (
```

Comment:
marshmallow allows you to pass strings for UUIDs, provided they are valid UUIDs. It would be good to preserver this behaviour.

Could we be cheeky and do something like:

```python
class SingleFilterValueField(fields.Field):

    def _serialize(self, value, attr, obj, **kwargs):
        if obj.by in {SingleFilterBy.PROJECT_ID, SingleFilterBy.RUN_ID}:
            field = fields.UUID()
        elif obj.by = SingleFilterBy.EXPERIMENT_ID:
            field = fields.Integer()
        elif obj.by = SingleFilterBy.DELETED_AT:
            field = fields.DateTime()
        # etc. for tag, param, metric
        return field._serialize(value, attr, obj, **kwargs)
```

Reply:
I had thought about doing something like this, but I worried about breaking conceptual encapsulation? Even though, to be fair, I don't understand why they would consider an explicit call to `_serialize` (or `_deserialize`) a bad thing.

If you believe it's okay to call `_serialize` I am happy to go for it

Reply:
I think it's fine to use it here, especially since the API to `_serialize` is stable / supported (we're implementing it ourselves for `SingleFilterValueField`.